crates/rooch/src/commands/init.rs

```diff
@@ -109,19 +113,37 @@ impl CommandAction<String> for Init {
                     Err(error) => return Err(RoochError::GenerateKeyError(error.to_string())),
                 };
 
-                let (new_address, phrase, key_pair_type) =
-                    keystore.generate_and_add_new_key(KeyPairType::RoochKeyPairType, None, None)?;
+                let password = if self.password_required == Some(false) {
+                    // Use an empty password if not required
+                    String::new()
+                } else {
+                    // Prompt for a password if required
+                    rpassword::prompt_password("Enter a password to encrypt the keys in the rooch keystore. Press return to have an empty value: ").unwrap()
+                };
+                println!("Your password is {}", password);
+
+                let result = keystore.generate_and_add_new_key(
+                    KeyPairType::RoochKeyPairType,
+                    None,
+                    None,
+                    Some(password),
+                )?;
                 println!(
-                    "Generated new keypair for address with type {:?} [{new_address}]",
-                    key_pair_type.type_of()
+                    "Generated new keypair for address with type {:?} [{}]",
+                    result.result.key_pair_type.type_of(),
+                    result.address
                 );
-                println!("Secret Recovery Phrase : [{phrase}]");
+                println!("Secret Recovery Phrase : [{}]", result.result.mnemonic);
                 let dev_env = Env::new_dev_env();
                 let active_env_alias = dev_env.alias.clone();
                 ClientConfig {
                     keystore,
                     envs: vec![env, dev_env],
-                    active_address: Some(new_address),
+                    password: Some(result.result.encryption.hashed_password),
+                    nonce: Some(result.result.encryption.nonce.encode_hex()),
+                    ciphertext: Some(result.result.encryption.ciphertext.encode_hex()),
```

Comment:
`keystore` file should be an independent file, not part of client_config.

Reply:
Removed.